PyInstaller/utils/win32/icon.py

```diff
@@ -188,16 +188,12 @@ def splitter(s):
 
     if len(srcpath) > 1:
         # More than one icon source given. We currently handle multiple icons by calling CopyIcons_FromIco(), which only
-        # allows .ico. In principle we could accept a mix of .ico and .exe, but it would complicate things. If you need
-        # it, submit a pull request.
+        # allows .ico, but will convert to that format if needed.
         #
         # Note that a ",index" on a .ico is just ignored in the single or multiple case.
         srcs = []
         for s in srcpath:
-            e = os.path.splitext(s[0])[1]
-            if e.lower() != '.ico':
-                raise ValueError('Multiple icons supported only from .ico files')
-            srcs.append(s[0])
+            srcs.append(normalize_icon_type(s[0], ("ico",), "ico", config.CONF["workpath"]))
```

Comment:
In the edge case of user passing a mixture of exe-based and image-based icons, the exe one will now raise a Pillow-conversion error. Perhaps we could keep the extension check here - and if it is exe, raise a `ValueError("Multiple icons supported only from image files.")` or something to that effect.)

On the other hand, the conversion error is also correct, so perhaps we can worry about this if it ever becomes a problem.